src/Utils.js

```diff
@@ -218,13 +218,14 @@ export function withSetStateAllowed(fn) {
     cleanup = true;
     global.document = {};
   }
-  fn();
+  const response = fn();
```

Comment:
```js
return try { fn(); } finally {
  if (cleanup) {
      // This works around a bug in node/jest in that developers aren't able to
      // delete things from global when running in a node vm.
      global.document = undefined;
      delete global.document;
    }
};
```

(this is arguably what it should have been doing all along, in case `fn` throws)